ember-amount-input/src/components/amount-input.ts

```diff
@@ -0,0 +1,180 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import './amount-input.css';
+
+const KEY_CODE_E = 69;
+const KEY_CODE_FULLSTOP = 190;
+const KEY_CODE_COMMA = 188;
+
+export interface AmountInputArgs {
+  /**
+   * The currency displayed in the input
+   */
+  currency?: string;
+
+  /**
+   * Disables the input
+   */
+  disabled?: boolean;
+
+  /**
+   * A custom class applied on the input
+   */
+  inputClass?: string;
+
+  /**
+   * A custom id applied on the input
+   */
+  inputId?: string;
+
+  /**
+   * Specifies the minimum value for the input field
+   */
+  min?: number;
+
+  /**
+   * Specifies the maximum value for the input field
+   */
+  max?: number;
+
+  /**
+   * Specifies the number of decimals to use for the amount value
+   */
+  numberOfDecimal?: number;
+
+  /**
+   * The placeholder displayed in the input
+   */
+  placeholder?: string;
+
+  /**
+   * Specifies if the input field should be read-only
+   */
+  readonly?: boolean;
+
+  /**
+   * Specifies the number intervals for the input field
+   */
+  step?: number;
+
+  /**
+   * The callback function triggered when the input value is updated
+   */
+  update: (value: string) => void;
```

Comment:
Why does it take a `string` as input? Shouldn't it be a `number`? Or it will be processed as a `number` later on?

Reply:
Each time we call the `update` function argument, we pass the new input value as a `string` (and not as a `number`). This is the case in `onFocusOut` ([here](https://github.com/qonto/ember-amount-input/blob/5098b99d91e17b327ceb38bd55b289faaae291fd/ember-amount-input/src/components/amount-input.ts#L162)) and in `onInput` ([here](https://github.com/qonto/ember-amount-input/blob/5098b99d91e17b327ceb38bd55b289faaae291fd/ember-amount-input/src/components/amount-input.ts#L152)).
Right now, with the current implementation, if a user needs to work with a `number` (and not a `string`, the type we return), he has to parse it by himself.

Reply:
I think we can refactor this as part of the issue I [mentioned here](https://github.com/qonto/ember-amount-input/pull/610#discussion_r1358479039), what do you think?